File: BI-TJV/semestral/rest-server/src/main/java/wrzecond/entity/TjvOrderFood.java
package wrzecond.entity;

import lombok.*;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@EqualsAndHashCode(callSuper = true)
@AllArgsConstructor
@NoArgsConstructor
public class TjvOrderFood extends TjvEntity {

    @ManyToOne
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "order_id", nullable = false)
    @Getter @Setter private TjvOrder order;

    @ManyToOne
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "food_id", nullable = false)
    @Getter @Setter private TjvFood food;

    @Getter @Setter private Timestamp datetime;

    @Column(nullable = false)
    @Getter @Setter private int count;

}

File: BI-KOT/semestral/rest-server/src/main/kotlin/wrzecond/entity/TjvOrderFood.kt
package wrzecond.entity

import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction
import java.sql.Timestamp
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne

@Entity
data class TjvOrderFood (
    @ManyToOne
    @OnDelete (action = OnDeleteAction.CASCADE)
    @JoinColumn (name = "order_id", nullable = false)
    val order: TjvOrder,

    @ManyToOne
    @OnDelete (action = OnDeleteAction.CASCADE)
    @JoinColumn (name = "food_id", nullable = false)
    val food: TjvFood,

    @Column(nullable = false) val datetime: Timestamp,
    @Column(nullable = false) val count: Int = 0,
    override val id: Int = 0
) : TjvEntity()
